### ISSUE TITLE
Cards show no activities

### ISSUE THREAD

**Issue by erl4nc** [RestyaPlatform/board#2666]
After upgrading to v0.6.7, some users saw no card comments or activities when they opened any card.  This behaviour persisted even after they deleted browser cache, logged out and logged back in.  If they clicked on the "comment" or "activities" button then comments/activities showed up. However, closing the card and re-opening it reverted the card back to showing no activities and users would have to click the buttons again each time they open a card.

## Expected Behavior
When a card is opened, previously selected card activities/comments setting is restored.

## Current Behavior
No activities or comments show when a card is opened even if the card has comments and activities.

## Possible Solution
Only some users experienced this.  When they did, postgres log showed:
`ERROR:  OFFSET must not be negative`
`STATEMENT:  SELECT row_to_json(d) FROM (SELECT al.*, u.username, u.profile_picture_path, u.initials, u.full_name, c.description, c.name as card_name FROM activities_listing al LEFT JOIN users u ON al.user_id = u.id LEFT JOIN cards c on al.card_id = c.id WHERE al.board_id = $1 AND al.card_id = $2 AND al.type = $3 ORDER BY freshness_ts DESC, depth ASC LIMIT 20  offset -20) as d`

and web server log showed the request being sent as
`api/v1/boards/25/lists/382/cards/1958/activities.json?view=modal_card&mode=comment&page=0&...`

`page=0` was the culprit. I was personally never able to reproduce this behaviour.  My requests were always `page=1`.

I found no apparent reason why a request with `page=0` was generated so I am not certain there actually is a bug in javascript.  Instead, I added a check in r.php, around line 1336.

```
if (isset($r_resource_filters['page'])) {
        $offset_val = ($r_resource_filters['page'] - 1) * PAGING_COUNT;
        $construct_offset = ' offset ' . $offset_val;
}
```

becomes
```
if (isset($r_resource_filters['page'])) {
	$offset_val = ($r_resource_filters['page'] - 1) * PAGING_COUNT;
	if ($offset_val < 0) {
		$offset_val = 0;
	}
	$construct_offset = ' offset ' . $offset_val;
}
```

Please advise if you see any harm in the php change.

**Comment by baskar383** [comment]
@erl4nc 
Thank you, We checked the issue, we are not able to reproduce.  we already faced these issues in Restyboard (v0.6.6) and we fixed this issue in Restyboard (v0.6.7),  Anyhow we updated the code as in r.php (https://github.com/RestyaPlatform/board/commit/474182bd2088bbe81e839505b0601497c0839533) as your wish. Kindly update your  Restyboard version screenshot you can get it to form this URL ```http://{RestyaboardURL}#/about```

**Comment by erl4nc** [comment]
This particular server was upgraded from v0.6.5 to v0.6.7.  It was never on v0.6.6. If it was indeed a confirmed defect in v0.6.6, I suspect the defect is still around in v0.6.7 though the change in r.php should take care of it.
![image](https://user-images.githubusercontent.com/33263752/60033210-0619f480-965d-11e9-82a0-b99db39b4c33.png)
